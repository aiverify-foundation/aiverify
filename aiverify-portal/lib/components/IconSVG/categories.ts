import { IconName } from '.';

export const lineIcons = [
  IconName.Alert,
  IconName.ArrowRight,
  IconName.ArrowLeft,
  IconName.Pencil,
  IconName.Asterisk,
  IconName.Book,
  IconName.BurgerMenu,
  IconName.ChatBubbleWide,
  IconName.ChatBubbles,
  IconName.CircleArrowRight,
  IconName.CircleArrowLeft,
  IconName.Close,
  IconName.Document,
  IconName.File,
  IconName.LayoutColumns,
  IconName.LayoutWtf,
  IconName.LightSun,
  IconName.List,
  IconName.Maximize,
  IconName.Minimize,
  IconName.OutlineBox,
  IconName.Pencil,
  IconName.Plus,
  IconName.Reset,
  IconName.RunCookbook,
  IconName.Square,
  IconName.Table,
  IconName.TalkBubbles,
  IconName.Warning,
  IconName.WideArrowUp,
  IconName.WideArrowDown,
  IconName.WideArrowLeft,
  IconName.WideArrowRight,
];

export const fillIcons = [
  IconName.Bell,
  IconName.CheckList,
  IconName.DarkMoon,
  IconName.Database,
  IconName.HistoryClock,
  IconName.LightBulb,
  IconName.Lightning,
  IconName.MoonPromptTemplate,
  IconName.Ribbon,
  IconName.SolidBox,
  IconName.Spacesuit,
  IconName.Tools,
  IconName.OpenedBook,
  IconName.OpenedEmptyBook,
  IconName.Plug,
];

export const colorIcons = [IconName.Folder, IconName.FolderForChatSessions];
