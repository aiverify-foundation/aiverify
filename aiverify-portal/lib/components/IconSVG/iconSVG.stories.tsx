import type { Meta, StoryObj } from '@storybook/react';
import { Icon, IconName } from '.';
import { fillIcons, lineIcons } from './categories';

const meta: Meta<typeof Icon> = {
  title: 'Components/SVG Icons',
  component: Icon,
  parameters: {
    controls: {
      sort: 'requiredFirst',
      expanded: true,
    },
  },
  argTypes: {
    name: {
      control: 'select',
      options: Object.values(IconName).sort(),
      description: 'Name of the icon',
    },
    size: {
      control: 'number',
      description: 'Size of the icon in CSS px',
    },
    svgClassName: {
      control: 'text',
      description: 'Class name for the SVG',
      table: {
        type: { summary: 'string' },
        disable: true,
      },
    },
    role: {
      table: {
        disable: true,
      },
    },
    ariaLabel: {
      table: {
        disable: true,
      },
    },
    style: {
      table: {
        disable: true,
      },
    },
    onMouseDown: {
      table: {
        disable: true,
      },
    },
    onClick: {
      table: {
        disable: true,
      },
    },
  },
};

export default meta;
type Story = StoryObj<typeof Icon>;

export const Line_Icon: Story = {
  argTypes: {
    name: {
      control: 'select',
      options: lineIcons.sort(),
      description: 'Name of the icon',
    },
  },
  args: {
    name: IconName.Alert,
    size: 50,
    svgClassName: 'stroke-primary-900 dark:stroke-primary-50',
    onClick: () => alert('Icon clicked'),
  },
};

export const Fill_Icon: Story = {
  argTypes: {
    name: {
      control: 'select',
      options: fillIcons.sort(),
      description: 'Name of the icon',
    },
  },
  args: {
    name: IconName.Bell,
    size: 50,
    svgClassName: 'fill-primary-900 dark:fill-primary-50',
    onClick: () => alert('Icon clicked'),
  },
};

export const Color_Icon: Story = {
  args: {
    name: IconName.Folder,
    size: 50,
    svgClassName: 'fill-primary-900 dark:fill-primary-50',
    onClick: () => alert('Icon clicked'),
  },
  argTypes: {
    name: {
      control: 'select',
      options: [IconName.Folder, IconName.FolderForChatSessions],
      description: 'Name of the icon',
    },
  },
};
