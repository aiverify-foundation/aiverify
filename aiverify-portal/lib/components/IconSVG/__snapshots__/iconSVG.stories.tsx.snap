// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/SVG Icons Color_Icon smoke-test 1`] = `
<div style="position: relative; width: 100%; height: 300px; display: flex; align-items: center; justify-content: center;">
  <div class="_icon_wrapper_1j07s_1 _pointer_effect_1j07s_17"
       style="width: 50px; height: 50px;"
  >
    <svg version="1.1"
         height="50"
         width="50"
         viewbox="0 0 24 24"
    >
      <g transform="translate(0 -1028.4)">
        <path d="m2 1033.4c-1.1046 0-2 0.9-2 2v14c0 1.1 0.89543 2 2 2h20c1.105 0 2-0.9 2-2v-14c0-1.1-0.895-2-2-2h-20z"
              fill="#2980b9"
        >
        </path>
        <path d="m3 1029.4c-1.1046 0-2 0.9-2 2v14c0 1.1 0.8954 2 2 2h11 5 2c1.105 0 2-0.9 2-2v-9-3c0-1.1-0.895-2-2-2h-2-5-1l-3-2h-7z"
              fill="#2980b9"
        >
        </path>
        <path d="m23 1042.4v-8c0-1.1-0.895-2-2-2h-11-5-2c-1.1046 0-2 0.9-2 2v8h22z"
              fill="#bdc3c7"
        >
        </path>
        <path d="m2 1033.4c-1.1046 0-2 0.9-2 2v6 1 6c0 1.1 0.89543 2 2 2h20c1.105 0 2-0.9 2-2v-6-1-6c0-1.1-0.895-2-2-2h-20z"
              fill="#3498db"
        >
        </path>
      </g>
    </svg>
  </div>
</div>
`;

exports[`Components/SVG Icons Fill_Icon smoke-test 1`] = `
<div style="position: relative; width: 100%; height: 300px; display: flex; align-items: center; justify-content: center;">
  <div class="_icon_wrapper_1j07s_1 _pointer_effect_1j07s_17"
       style="width: 50px; height: 50px;"
  >
    <svg xmlns="http://www.w3.org/2000/svg"
         width="50"
         height="50"
         viewbox="0 0 24 24"
         fill="none"
         stroke="#000000"
         stroke-width="2"
         stroke-linecap="round"
         stroke-linejoin="round"
         class="fill-primary-900 dark:fill-primary-50"
    >
      <circle cx="12"
              cy="12"
              r="10"
      >
      </circle>
      <line x1="12"
            y1="8"
            x2="12"
            y2="12"
      >
      </line>
      <line x1="12"
            y1="16"
            x2="12.01"
            y2="16"
      >
      </line>
    </svg>
  </div>
</div>
`;

exports[`Components/SVG Icons Line_Icon smoke-test 1`] = `
<div style="position: relative; width: 100%; height: 300px; display: flex; align-items: center; justify-content: center;">
  <div class="_icon_wrapper_1j07s_1 _pointer_effect_1j07s_17"
       style="width: 50px; height: 50px;"
  >
    <svg xmlns="http://www.w3.org/2000/svg"
         width="50"
         height="50"
         viewbox="0 0 24 24"
         fill="none"
         stroke="#000000"
         stroke-width="2"
         stroke-linecap="round"
         stroke-linejoin="round"
         class="stroke-primary-900 dark:stroke-primary-50"
    >
      <circle cx="12"
              cy="12"
              r="10"
      >
      </circle>
      <line x1="12"
            y1="8"
            x2="12"
            y2="12"
      >
      </line>
      <line x1="12"
            y1="16"
            x2="12.01"
            y2="16"
      >
      </line>
    </svg>
  </div>
</div>
`;
